# Ejemplo de la inversa de una matriz

Se llama matriz inversa de una matriz cuadrada A, y se expresa A-1, a la única matriz que cumple que:
*A·A-1 = I = A-1·A*


In [1]:
import numpy
import scipy.linalg as sl
from numpy import array
A=array ([[1, 2], [3, 4]])
A

array([[1, 2],
       [3, 4]])

In [4]:
det= sl.inv(A)

In [5]:
det

array([[-2. ,  1. ],
       [ 1.5, -0.5]])

**Documentación**      
Es decir, la matriz inversa de A es la única matriz que al multiplicarla por ella obtenemos la matriz identidad del orden correspondiente.

La matriz inversa no siempre existe, para que exista, es condición necesaria y suficiente que el determinante de la matriz sea distinto de cero    
**Formula**
Para calcular la inversa de una matriz se multiplica $A* A ^-1$ = $ A^-1*A$

**Solución teórica**   
La inversa de una matriz se realizo con la formula     
  A∗A−1  =  A−1∗A    

  
**Solución a mano**
 Construir una matriz del tipo M = (A | I)  
 [1, 2]  = A,     
 [3, 4]  
La matriz identidad se escribe de la siguente manera:
 [1, 0]  = I,     
 [0, 1]  

Y se pone una linea imaginaria para diferenciar la matriz dibujada y la matriz identidad, cuando se tiene una matriz pequeña no hay problema para distinguir cual es cada una, a diferencia cuando la matriz es mayor, se complica visualmente.
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|3 | 4| 0 | 1|

La matriz A se convertira en una identidad, con transformación, que de igual manera afecta a la matriz identidad original, y lo que resulte en donde se encontraba la matriz identidad, esa sera la inversa de la matriz.    
+ Primero se tiene que hacer '1'  en la cordenada (0, 0), en este caso el uno ya esta, ahora '0' en la coordenada (1, 0)


- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|**3** | 4| 0 | 1|
**-3F1+F2**

Operación  
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|-3 | -6| -3 | 0|
|**3** | 4| 0 | 1|
**0** |**-2** |**-3** | **1**|

Despues de la operación la matriz queda asi:
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|0| -2| -3 | 1|

El siguiente paso es hacer '1' el '-2' con coordenada (1, 1); que se logra con 1/2 y para que quede positivo se le pone un '-', de esta manera la multiplicacion de dos menos, da de resultado mas. (-) * (-) = (+)
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|0| -2| -3 | 1|
-1/2F2
Toda la fila 2 se multiplica por (-1/2)

Operación aplicada 
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|0| 1| 3/2 | -1/2|

Para terminar solo falta hacer '0' la coordenada (0, 1), el '2'.
Lo que se hace es, Multiplicar -2* F2 + F1
- | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 2| 1 | 0|
|0| 1| 3/2 | -1/2|

Operación

- | Matriz A | -|Identidad
-- | -- | -- | -- |
|0 | -2 |  -6/3 | 1|
|1| 2|1 | 0|
 
 
 y queda:
 - | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 0| -2 | 1|
|0| 1| 3/2 | -1/2|
 En fracciones
 
 y queda:
 - | Matriz A | -|Identidad
-- | -- | -- | -- |
|1 | 0| -2 | 1|
|0| 1| 1.5 | -.5|
con decimales